scripts/kind-e2e/e2e.sh

```diff
@@ -117,34 +147,95 @@ function test_connection() {
+
+        attempt_counter=$(($attempt_counter+1))
+        sleep 10
     done
+    echo "Connection test was successful!"
+}
+
+function update_subm_pods() {
+    echo Removing submariner engine pods...
+    kubectl get pods -n submariner -l app=submariner-engine | awk 'FNR > 1 {print $1}' | xargs kubectl delete pods -n submariner
+    kubectl wait --for=condition=Ready pods -l app=submariner-engine -n submariner --timeout=60s
+    echo Removing submariner route agent pods...
+    kubectl get pods -n submariner -l app=submariner-routeagent | awk 'FNR > 1 {print $1}' | xargs kubectl delete pods -n submariner
+    kubectl wait --for=condition=Ready pods -l app=submariner-routeagent -n submariner --timeout=60s
+}
+
+function enable_logging() {
+    kubectl config use-context cluster1
+    if kubectl rollout status deploy/kibana > /dev/null 2>&1; then
+        echo Elasticsearch stack already installed, skipping...
+    else
+        echo Installing Elasticsearch...
+        es_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' cluster1-control-plane | head -n 1)
+        kubectl apply -f ./kind-e2e/logging/elasticsearch.yaml
+        kubectl apply -f ./kind-e2e/logging/filebeat.yaml
+        echo Waiting for Elasticsearch to be ready...
+        kubectl wait --for=condition=Ready pods -l app=elasticsearch --timeout=300s
+        for i in 2 3; do
+            kubectl config use-context cluster${i}
+            kubectl apply -f ./kind-e2e/logging/filebeat.yaml
+            kubectl set env daemonset/filebeat -n kube-system ELASTICSEARCH_HOST=${es_ip} ELASTICSEARCH_PORT=30000
+        done
+    fi
 }
 
 function cleanup {
-  echo "Cleanup..."
-  for i in 1 2 3; do kind delete cluster --name=cluster${i}; done
+  for i in 1 2 3; do
+
+    if [[ $(kind get clusters | grep cluster${i} | wc -l) -gt 0  ]]; then
+        kind delete cluster --name=cluster${i};
+    fi
+    done
+
+    if [[ $(docker ps -qf status=exited | wc -l) -gt 0 ]]; then
+        echo Cleaning containers...
+        docker ps -qf status=exited | xargs docker rm -f
+    fi
+    if [[ $(docker images -qf dangling=true | wc -l) -gt 0 ]]; then
+        echo Cleaning images...
+        docker images -qf dangling=true | xargs docker rmi -f
+    fi
+#    if [[ $(docker images -q --filter=reference='submariner*:local' | wc -l) -gt 0 ]]; then
+#        docker images -q --filter=reference='submariner*:local' | xargs docker rmi -f
+#    fi
+    if [[ $(docker volume ls -qf dangling=true | wc -l) -gt 0 ]]; then
+        echo Cleaning volumes...
+        docker volume ls -qf dangling=true | xargs docker volume rm -f
+    fi
 }
 
-if [[ $1 = -c ]]; then
+if [[ $1 = clean ]]; then
     cleanup
     exit 0
 fi
 
-if [[ $1 != -k ]]; then
+if [[ $1 != keep ]]; then
     trap cleanup EXIT
 fi
 
-helm init --client-only
-helm repo add submariner-latest https://releases.rancher.com/submariner-charts/latest
+echo Starting a job with status: $1, k8s version: $2, monitor: $3.
```

Comment:
nit, I would omit "a job",  because in some cases it will be a devel env :-)

Reply:
Yes good suggestion.